pkg/kubelet/gpu/nvidia/nvidia_gpu_manager.go

```diff
@@ -190,25 +204,61 @@ func (ngm *nvidiaGPUManager) updateAllocatedGPUs() {
 // family name. Need to support NVML in the future. But we do not need NVML until
 // we want more features, features like schedule containers according to GPU family
 // name.
-func (ngm *nvidiaGPUManager) discoverGPUs() error {
+func (ngm *nvidiaGPUManager) discoverGPUs(devicesFile string, devDir string) error {
+	found, err := expectedGPUs(devicesFile)
+	if err != nil || !found {
+		return err
```

Comment:
It was not directly obvious to me that this could be a return path in normal (non error circumstances). I would recommend splitting this if condition into two blocks, one for `err != nil` and another for the other non-error case.